pkg/executor/aggregate/agg_hash_partial_worker.go

```diff
@@ -107,7 +104,55 @@ func (w *HashAggPartialWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitG
 	}
 }
 
-func (w *HashAggPartialWorker) updatePartialResult(ctx sessionctx.Context, sc *stmtctx.StatementContext, chk *chunk.Chunk, _ int) (err error) {
+func newPartialResults(rowNum int, finalConcurrency int) [][][]aggfuncs.PartialResult {
+	partialResults := make([][][]aggfuncs.PartialResult, finalConcurrency)
+	avgRowNum := rowNum / finalConcurrency
+	for i := 0; i < finalConcurrency; i++ {
+		partialResults[i] = make([][]aggfuncs.PartialResult, 0, avgRowNum)
+	}
+	return partialResults
+}
+
+// It will get partial results for all final workers according to the group by key
+func (w *HashAggPartialWorker) getPartialResultsForAllFinalWorker(_ *stmtctx.StatementContext, groupKey [][]byte, mapper []AggPartialResultMapper, finalConcurrency int) ([]int, [][][]aggfuncs.PartialResult) {
+	n := len(groupKey)
+	partialResults := newPartialResults(n, finalConcurrency)
```

Comment:
Can we reuse the slice? It can reduce the frequency of memory allocation.

Reply:
> Can we reuse the slice? It can reduce the frequency of memory allocation.

okk, I will reuse this slice.